deps/db/test/logseq/db/frontend/rules_test.cljs

```diff
@@ -19,22 +23,23 @@
                         :build/properties {:foo "bar"}}}]})]
 
     (is (= ["Page"]
-           (->> (q-with-rules '[:find (pull ?b [:block/title]) :where (has-page-property ?b :user.property/foo)]
+           (->> (q-with-rules '[:find (pull ?b [:block/title]) :where (has-property ?b :user.property/foo)]
                               @conn)
                 (map (comp :block/title first))))
-        "has-page-property returns result when page has property")
+        "has-property returns result when block has property")
     (is (= []
-           (->> (q-with-rules '[:find (pull ?b [:block/title]) :where (has-page-property ?b :user.property/foo2)]
+           (->> (q-with-rules '[:find (pull ?b [:block/title]) :where (has-property ?b :user.property/foo2)]
                               @conn)
                 (map (comp :block/title first))))
-        "has-page-property returns no result when page doesn't have property")
+        "has-property returns no result when block doesn't have property")
     (is (= [:user.property/foo]
-           (q-with-rules '[:find [?p ...]
-                           :where (has-page-property ?b ?p) [?b :block/title "Page"]]
-                         @conn))
-        "has-page-property can bind to property arg")))
+           (->> (q-with-rules '[:find [?p ...]
+                                :where (has-property ?b ?p) [?b :block/title "Page"]]
+                              @conn)
+                (remove-built-in-properties conn)))
```

Comment:
This look like a regression caused by making so many built-in attributes act as properties. Removing this fn shows the results to be `[:user.property/foo :block/updated-at :block/created-at :block/title :block/type]`. None of those :block attributes are understood to be properties by us or the users so this will cause bugs with using this rule

Reply:
The query lists all the properties that this page has, it makes sense to me to list those :block attributes as its properties. 
Those attributes are always displayed on the UI, so it should only affect users' workflows slightly.
I can update the rule to filter out db attributes, but considering it is such a rare case that users will use `has-property` to list all the properties instead of querying which block has a property or a property with a specific value, I don't think it's worth making the change.